src/api_client/api_client.py

```diff
@@ -20,18 +24,91 @@ def get_exchange_rates(self, currency_pair, interval, start_time, end_time):
+        if currency_pair_exists:
+            exchange_rate = self.client.get_symbol_ticker(symbol=f'{currency_to_buy}{currency_to_sell}')['price']
+            return float(exchange_rate)
+        else:
+            exchange_rate = self.client.get_symbol_ticker(symbol=f'{currency_to_sell}{currency_to_buy}')['price']
+            return 1 / float(exchange_rate)
 
-    def trade(self, currency_to_buy, currency_to_sell, action, quantity):
-        response = self.client.create_order(symbol=f'{currency_to_buy}{currency_to_sell}',
-                                            quantity=round(quantity),
-                                            side=action,
-                                            type='MARKET')
-        return response['orderId']
+    def trade(self, currency_to_buy, currency_to_sell, action, amount, data=None):
+        precision = self.get_amount_precision(currency_pair=f'{currency_to_buy}{currency_to_sell}')
+        response = self.client.create_test_order(symbol=f'{currency_to_buy}{currency_to_sell}',
+                                                 quantity=f"%.{precision}f" % amount,
+                                                 side=action,
+                                                 type=enums.ORDER_TYPE_MARKET)
+        return response
+
+    def get_balances(self):
+        """
+        Get all the balances with values of more than 0 of the account.
+        :return: pd.DataFrame
+        """
+        info = self.client.get_account()
+        portfolio = pd.DataFrame(info['balances'])
+        portfolio = portfolio.set_index('asset')
+        portfolio = portfolio.astype(float)
+        return portfolio[portfolio['free'] > 0]
+
+    def get_commission(self, currency_pair=f'{config.CURRENCY_TO_BUY}{config.CURRENCY_TO_SELL}'):
+        fees = self.client.get_trade_fee(symbol=currency_pair)
+        return float(fees[0]['takerCommission'])
+
+    def get_amount_precision(self, currency_pair=f'{config.CURRENCY_TO_BUY}{config.CURRENCY_TO_SELL}'):
+        asset_info = self.client.get_symbol_info(currency_pair)
+        step_size = float(asset_info['filters'][2]['stepSize'])
+        return int(np.log10(round(1 / step_size)))
+
+    def get_minimum_amount(self, currency_to_buy, currency_to_sell):
+        asset_info = self.client.get_symbol_info(f'{currency_to_buy}{currency_to_sell}')
+        if asset_info['filters'][3]['applyToMarket']:
+            amount_usdt = float(asset_info['filters'][3]['minNotional'])
+        else:
+            amount_usdt = float(asset_info['filters'][5]['minNotional'])
+        price = self.get_current_exchange_rate(currency_to_buy=currency_to_buy, currency_to_sell=currency_to_sell)
+        return amount_usdt / price
+
+
+class BinanceBackTestClient(BinanceClient):
+    def __init__(self, api_key, api_secret, **kwargs):
+        super().__init__(api_key, api_secret, **kwargs)
+        self.balances = pd.DataFrame(
```

Comment:
Instead of hardcoding it, we might need to create a function called `set_balance` that it's going to be triggered to update the backtesting balance, right? Otherwise it will always read this fixed value.